File: articles/what-the-meta-analyses-say-about-grit/fortran/fisher_z_demo.f90
program fisher_z_demo
  implicit none
  real :: r, z, e, back_r

  r = 0.22
  z = 0.5 * log((1.0 + r) / (1.0 - r))
  e = exp(2.0 * z)
  back_r = (e - 1.0) / (e + 1.0)

  print *, "r:", r
  print *, "Fisher z:", z
  print *, "Back-transformed r:", back_r
end program fisher_z_demo

File: articles/what-the-meta-analyses-say-about-grit/cpp/fisher_z_demo.cpp
#include <cmath>
#include <iostream>

double fisher_z(double r) {
    return 0.5 * std::log((1.0 + r) / (1.0 - r));
}

double inverse_fisher_z(double z) {
    double e = std::exp(2.0 * z);
    return (e - 1.0) / (e + 1.0);
}

int main() {
    double r = 0.22;
    double z = fisher_z(r);
    std::cout << "r: " << r << "\n";
    std::cout << "Fisher z: " << z << "\n";
    std::cout << "Back-transformed r: " << inverse_fisher_z(z) << "\n";
    return 0;
}
